### ISSUE TITLE
[Style] [table, table-column] Table inside space with fill props the header width is growing to infinity

### ISSUE THREAD

**Issue by okyaneka** [element-plus/element-plus#9652]
Bug Type: **`Style`**

## Environment

- Vue Version: `3.2.37`
- Element Plus Version: `2.2.14`
- Browser / OS: `Firefox 103.0.2 Ubuntu 22.04`
- Build Tool: `Vite`

### Related Component

- `el-table`
- `el-table-column`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBtc2cgPSByZWYoJ0hlbGxvIFdvcmxkIScpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuPGVsLXNwYWNlIGZpbGw+XG4gIDxlbC10YWJsZT5cbiAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJyYW5rXCIgbGFiZWw9XCIjXCIgLz5cbiAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJuYW1lXCIgbGFiZWw9XCJOYW1lXCIgLz5cbiAgICA8ZWwtdGFibGUtY29sdW1uIHByb3A9XCJzY29yZVwiIGxhYmVsPVwiU2NvcmVcIiAvPlxuICA8L2VsLXRhYmxlPlxuPC9lbC1zcGFjZT5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)

### Steps to reproduce

```vue
<el-space fill>
  <el-table>
    <el-table-column prop="rank" label="#" />
    <el-table-column prop="name" label="Name" />
    <el-table-column prop="score" label="Score" />
  </el-table>
</el-space>
```

### What is Expected?

table width is not growing to infinity

### What is actually happening?

table width is growing to infinity

### Additional comments

_(empty)_

**Comment by GODLiangCY** [comment]
Maybe we can add a CSS style `max-width: 100%` to `.el-space` class when `fill` is set to true?

**Comment by okyaneka** [comment]
yes, it prevent the table width to grow. I add `width: 100%` to make initial table size to 100% of width

**Comment by Dsaquel** [comment]
Fixed in newer versions.
